scalameta/io/shared/src/main/scala/scala/meta/io/AbsolutePath.scala

```diff
@@ -1,29 +1,27 @@
 package scala.meta.io
 
-import scala.meta.internal.io.PlatformIO._
+import java.io._
+import org.scalameta.data._
+import scala.meta.internal.io.PlatformIO
 
-// sealed abstract to ensure that AbsolutePath.apply constructor is used.
-sealed abstract class AbsolutePath(val absolute: String) extends Serializable {
```

Comment:
Good point. I will see what we can do in `@data` to make the constructor private.